src/panels/developer-tools/state/developer-tools-state.js

```diff
@@ -296,7 +297,7 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
         return hass.states[key];
       })
       .filter(function (value) {
-        if (!value.entity_id.includes(_entityFilter.toLowerCase())) {
+        if (fuzzyFilter(_entityFilter.toLowerCase(), value.entity_id)) {
```

Comment:
This is the wrong way? I would expect `fuzzyFilter` to return `true` if the filter matches the word?

Reply:
Yeah, the naming is tricky. It would return true if it were a fuzzy _matcher_. But this is a _filter_. Returning "true" would mean "yes, please filter out this string".

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

True in a filter means return this item

Reply:
> Both vscode and chrome show what letters of the search results match your filter.

Ahh. Got it. That's a totally fair point -- _any_ sort of "fuzzy" or "smart" filtering in this context needs to be clear to the user about why a result was returned.

Would you be open to a follow-up PR that implements character-bolding, and perf tests? Or do you just feel like this isn't a good idea.

> I don't think sorting the table is something we should do, that would only work in a dedicated search dropdown?

The filter already performs a real-time [sort after filtering](https://github.com/home-assistant/frontend/blob/dev/src/panels/developer-tools/state/developer-tools-state.js#L349).

Reply:
I think there's some confusion here. I just edited my original comment for clarity. 

The logic does work out to the expected behavior because I think of "filter" as "a thing that filters _out_ mismatches rather than "a thing that allows matches through the filter".